### PR TITLE
Python 3 fixes and tests

### PR DESCRIPTION
**Pull request by petschki**

### CODE REVIEW

File: src/collective/collectionfilter/filteritems.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def get_filter_items(
     groupby_criteria = getUtility(IGroupByCriteria).groupby
     idx = groupby_criteria[group_by]['index']
     current_idx_value = request_params.get(idx)
-    if not getattr(current_idx_value, '__iter__', False):
+    if not isinstance(current_idx_value, list):
```
Comment (petschki) [inline src/collective/collectionfilter/filteritems.py]: strings do have `__iter__` in python 3 ... doh

Comment (thet) [reply src/collective/collectionfilter/filteritems.py]: the ``__iter__`` war probably meant to handle cases where ``current_idx_value`` was something iterable. I wonder if it wouldn't be better to do something like:
```
current_idx_value = current_idx_value or []  # handle None
if isinstance(current_idx_value, str):
    # do not expand a string to a list of chars
    current_idx_value = [current_idx_value]
else:
    try:
        current_idx_value = list(current_idx_value)
    except TypeError:
        # int and stuff.
        current_idx_value = [current_idx_value]
```

Comment (petschki) [reply src/collective/collectionfilter/filteritems.py]: yes ... this look more failsafe than simply checking for `list` ... I'll change that

**Comment by agitator** [comment]
awesome work, thanx!

**Pull request merged**